src/vault/authenticate.clj

```diff
@@ -132,3 +132,35 @@
            :content-type :json
            :accept :json
            :as :json})))))
+
+
+(defmethod authenticate* :aws
+  [client _ credentials]
+  (let [{:keys [api-path iam-http-request-method iam-request-url iam-request-body
+                iam-request-headers role]} credentials
```

Comment:
Since we're already talking about IAM authentication from context here, I don't think these keys need to be prefixed with `iam-`. For example, the LDAP auth credentials don't take `:ldap-username` and `:ldap-password`, Kubernetes takes `:jwt` rather than `:k8s-jwt`, etc.

Reply:
👍  Will make the change.